x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/deduplication_panel/utils.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiSelectableLIOption } from '@elastic/eui/src/components/selectable/selectable_option';
+
+export const getSelectableOptions = (
+  availableFields: string[],
+  selectedFields: string[],
+  showAllFields: boolean,
+  deduplicationEnabled: boolean
+): Array<EuiSelectableLIOption<object>> => {
```

Comment:
When calling this function, it doesn't feel intuitive which positional argument is what. Accepting an object as the argument would make that clearer IMO.

Reply:
Or perhaps it could accept a `domain` argument and a `showAllFields` boolean, that would also help a bit.

Reply:
Passing domain as an argument is a great suggestion, I've implemented it here 5b36a242b